elyra/pipeline/component_catalog.py

```diff
@@ -72,90 +70,165 @@ class ComponentCatalog(LoggingConfigurable):
-        elif operation == 'delete':
-            # Remove only the components most recently re-read
-            for component_id, component in updated_components.items():
-                self._cached_components.pop(component_id)
-        else:
-            # Replace all components in the cache
-            self._cached_components = updated_components
+    def _get_component_parser_class(self, platform_type: str) -> Optional[ComponentParser]:
+        """
+        Return the appropriate component parser based on the runtime platform type indicated
+        in the component catalog metadata instance
+        """
+        try:
+            parser = ComponentParser(platform_type=platform_type)
+        except Exception as e:
+            self.log.error(f"Could not get appropriate ComponentParser class: {e}. Skipping...")
+            return None
+
+        return parser
+
+    def _load_catalog_reader_class(self, catalog: Metadata, file_types: List[str]) \
+            -> Optional[ComponentCatalogConnector]:
+        """
+        Load the appropriate entrypoint class based on the schema name indicated in
+        the catalog Metadata instance and the file types associated with the component
+        parser in use
+        """
+        try:
+            catalog_reader = entrypoints.get_group_named('elyra.component.catalog_types').get(catalog.schema_name)
+            if not catalog_reader:
+                self.log.error(f"No entrypoint with name '{catalog.schema_name}' was found in group "
+                               f"'elyra.component.catalog_types' to match the 'schema_name' given in catalog "
+                               f"'{catalog.display_name}'. Skipping...")
+                return None
+
+            catalog_reader = catalog_reader.load()(file_types, parent=self.parent)
+        except Exception as e:
+            self.log.error(f"Could not load appropriate ComponentCatalogConnector class: {e}. Skipping...")
+            return None
+
+        return catalog_reader
+
+    def _read_component_catalog(self, catalog: Metadata) -> Dict[str, Component]:
+        """
+        Read a component catalog and return a dictionary of components indexed by component_id.
+
+        :param catalog: a metadata instances from which to read and construct Component objects
 
-        self._cache_last_updated = time.time()
+        :returns: a dictionary of component id to Component object for all read/parsed components
+        """
+        component_dict: Dict[str, Component] = {}
 
-    def _is_cache_expired(self) -> bool:
-        is_expired = True
-        if self._cache_last_updated:
-            now = time.time()
-            elapsed = int(now - self._cache_last_updated)
-            if elapsed < self.cache_ttl:
-                is_expired = False
+        # Assign component parser based on the runtime platform type
+        parser = self._get_component_parser_class(platform_type=catalog.metadata['runtime_type'])
```

Comment:
If we're going to use `platform_type` as the full type, we should extend `ComponentCatalogMetadata` with a `runtime_type` property - something like:
```python
@property
def runtime_type() -> RuntimeProcessorType:
    return RuntimeProcessorType.get_instance_by_name(self.metadata['runtime_type'])
```